packages/adminrouter/extra/src/lib/cache.lua

```diff
@@ -235,8 +235,20 @@ local function fetch_and_store_marathon_apps(auth_token)
        end
 
        local ports = task["ports"] --task host port mapping by default
-       if is_container_network(app) then
-         -- override with ports from the container's portMappings
+       -- task["ports"] will contain the host ports made available to the task:
+       -- see https://mesosphere.github.io/marathon/docs/ports.html#definitions,
+       -- note on the hostPort definitions.
+       -- AR, however, needs the container port to route the request to.
+       -- In "container" mode we find the container port out from portMappings array
+       -- for the case when container port is fixed (non-zero value specified).
+       -- When container port is specified as 0 it will be dynamically assigned
+       -- by Marathon and the actual container port will be the same as the host port:
+       -- https://mesosphere.github.io/marathon/docs/ports.html#random-port-assignment
+       -- We do not override it with container port from the portMappings array
+       -- in that case.
+       -- In "container/bridge" and "host" networking modes we need to use the
+       -- host port for routing (available via task's ports array)
+       if is_container_network(app) and app["container"]["portMappings"][1]["containerPort"] ~= 0 then
```

Comment:
instead of `["portMappings"][1]` do you really want `["portMappings"][portIdx]`?